SwanProjects/swanprojects/kernelmanager/kernelspecmanager.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) SWAN Development Team.
+# Author: Omar.Zapata@cern.ch 2021
+
+import json
+import os
+import shutil
+
+from jupyter_client.kernelspec import KernelSpecManager, NoSuchKernel
+from swanprojects.utils import (get_project_info, get_project_name,
+                                get_project_path, get_env_isolated)
+from traitlets import Unicode
+from traitlets.config import Configurable
+
```

Comment:
how does this (and jupyterlab in general) work with multiple windows? We're setting the project path just before doing any action that uses the kernels specs?

Reply:
When the file browser is changing the path, the path is set in the kernel manager, if the path is a project then the kernels are set.

In multiple windows, you can have multiple launchers in different projects and open for the projects notebooks with different kernels.

The problem is restoring the session or refreshing the browser, we don't have a session manager.